File: APP-ONTSTUDIO/src/api/lineage.ts
import { get, post } from './client';
import type { DataLineage, LineageImpactResult } from '@/types';

/**
 * 获取血缘图。
 * 后端路径：GET /v1/data/lineage?scope=xxx
 */
export async function getLineage(scope = 'all'): Promise<DataLineage> {
  const data = await get<DataLineage>('/v1/data/lineage', { scope });
  return data;
}

/**
 * 获取以指定节点为根的子树血缘。
 * 后端路径：GET /v1/data/lineage/{nodeId}
 */
export async function getLineageByNode(nodeId: string): Promise<DataLineage> {
  const data = await get<DataLineage>(`/v1/data/lineage/${nodeId}`);
  return data;
}

/**
 * 影响分析：修改某节点后受影响的上下游节点。
 * 后端路径：POST /v1/data/lineage/impact  body: { nodeId }
 */
export async function analyzeImpact(nodeId: string): Promise<LineageImpactResult> {
  return await post<LineageImpactResult>('/v1/data/lineage/impact', { nodeId });
}

File: TECH-DATA/app/api/v1/lineage.py
"""Lineage endpoints (V11-02)."""

from __future__ import annotations

from fastapi import APIRouter, Depends, Query, Request

from app.common.api_response import success
from app.common.context import RequestContext, request_context_dep
from app.deps import get_lineage_service
from app.lineage.schemas import ImpactAnalysisRequest
from app.lineage.service import LineageService

router = APIRouter(tags=["lineage"])


@router.get("/lineage", summary="获取血缘图（按 scope 过滤）")
async def get_lineage(
    request: Request,
    scope: str = Query(default="all"),
    ctx: RequestContext = Depends(request_context_dep),
    service: LineageService = Depends(get_lineage_service),
) -> dict:
    data = service.get_lineage(scope)
    return success(data.model_dump(mode="json"), trace_id=ctx.trace_id)


@router.get("/lineage/{node_id}", summary="获取以指定节点为根的子树血缘")
async def get_lineage_by_node(
    node_id: str,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: LineageService = Depends(get_lineage_service),
) -> dict:
    data = service.get_lineage_by_node(node_id)
    return success(data.model_dump(mode="json"), trace_id=ctx.trace_id)


@router.post("/lineage/impact", summary="影响分析")
async def analyze_impact(
    body: ImpactAnalysisRequest,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: LineageService = Depends(get_lineage_service),
) -> dict:
    data = service.analyze_impact(body.nodeId)
    return success(data.model_dump(mode="json"), trace_id=ctx.trace_id)
